database/service_db.go

```diff
@@ -52,59 +75,103 @@ func (d *LevelDBServiceDB) unmarshal(id string, value []byte) (*service.Service,
 func (d *LevelDBServiceDB) All() ([]*service.Service, error) {
 	var (
 		services []*service.Service
-		iter     = d.db.NewIterator(nil, nil)
+		iter     = d.db.NewIterator(util.BytesPrefix([]byte(idKeyPrefix)), nil)
 	)
-
 	for iter.Next() {
-		s, err := d.unmarshal(string(iter.Key()), iter.Value())
+		id := strings.TrimPrefix(string(iter.Key()), idKeyPrefix)
+		s, err := d.unmarshal(id, iter.Value())
 		if err != nil {
 			// NOTE: Ignore all decode errors (possibly due to a service
 			// structure change or database corruption)
 			if decodeErr, ok := err.(*DecodeError); ok {
 				logrus.WithField("service", decodeErr.ID).Warning(decodeErr.Error())
 				continue
 			}
+			iter.Release()
 			return nil, err
 		}
 		services = append(services, s)
 	}
 	iter.Release()
-	if err := iter.Error(); err != nil {
-		return nil, err
-	}
-
-	return services, nil
+	return services, iter.Error()
 }
 
 // Delete deletes service from database.
-func (d *LevelDBServiceDB) Delete(id string) error {
-	return d.db.Delete([]byte(id), nil)
+func (d *LevelDBServiceDB) Delete(idOrAlias string) error {
+	tx, err := d.db.OpenTransaction()
```

Comment:
Why opening a transaction if the get and delete function are not using it?
It should be tx.Get, tx.Delete, etc...
https://godoc.org/github.com/syndtr/goleveldb/leveldb#Transaction

Reply:
Sorry it was in rush, I've pushed a fix. now should be ok